opensearch-operator/pkg/helpers/helpers.go

```diff
@@ -380,3 +394,38 @@ func CalculateJvmHeapSize(nodePool *opsterv1.NodePool) string {
 
 	return nodePool.Jvm
 }
+
+func UpgradeInProgress(status opsterv1.ClusterStatus) bool {
+	componentStatus := opsterv1.ComponentStatus{
+		Component: "Upgrader",
+	}
+	_, found := FindFirstPartial(status.ComponentsStatus, componentStatus, GetByComponent)
+	return found
+}
+func ReplicaHostName(currentSts appsv1.StatefulSet, repNum int32) string {
+	return fmt.Sprintf("%s-%d", currentSts.ObjectMeta.Name, repNum)
+}
+
+func WorkingPodForRollingRestart(ctx context.Context, k8sClient client.Client, sts *appsv1.StatefulSet) (string, error) {
+	// Handle the simple case
+	if lo.FromPtrOr(sts.Spec.Replicas, 1) == sts.Status.UpdatedReplicas+sts.Status.CurrentReplicas {
```

Comment:
Since `lo.FromPtrOr(sts.Spec.Replicas, 1)` is used several times in this function, why not pull it into a variable? Should make it a bit easier to read